soco/core.py

```diff
@@ -1833,6 +1873,185 @@ def add_item_to_sonos_playlist(self, queueable_item, sonos_playlist):
+                ("UpdateID", update_id),
+                ("TrackList", track),
+                ("NewPositionList", position),
+            ])
+            change += int(response['QueueLengthChange'])
+            update_id = int(response['NewUpdateID'])
+        length = int(response['NewQueueLength'])
+
+        return change, length, update_id
+
+    def clear_sonos_playlist(self, object_id, update_id=0):
+        """ Clear all tracks from a Sonos playlist.
+        This is a convenience function for :py:meth:`reorder_sonos_playlist`.
+
+        Example:
+            ``obj.clear_sonos_playlist('SQ:0')``
+
+        Args:
+            object_id (str): The item_id of the Sonos playlist or
+                the Sonos playlist object
+                (:py:class:`~.soco.data_structures.DidlPlaylistContainer`).
+            update_id (int): Optional update counter for the object. If left
+                at the default of 0, it will be looked up.
+
+        Returns:
+            tuple: See :py:meth:`reorder_sonos_playlist`
+
+        Raises:
+            SoCoUPnPException: See :py:meth:`reorder_sonos_playlist`
+        """
+        sonos_pl = None
+        if type(object_id) is DidlPlaylistContainer:
+            sonos_pl = object_id
+        else:
+            for spl in self.get_sonos_playlists():
+                if spl.item_id == object_id:
+                    sonos_pl = spl
+        if sonos_pl is None:
+            raise ValueError('"{0}" is not a valid Sonos Playlist'
+                             ' item_id'.format(object_id))
+        num_tracks = self.browse(ml_item=sonos_pl).total_matches
+        tracks = ','.join([str(x) for x in range(num_tracks)])
+        return self.reorder_sonos_playlist(sonos_pl, tracks=tracks,
+                                           new_pos='', update_id=update_id)
+
+    def move_in_sonos_playlist(self, object_id, track, new_pos, update_id=0):
+        """ Move a track to a new position within a Sonos Playlist.
+        This is a convenience function for :py:meth:`reorder_sonos_playlist`.
+
+        Example::
+
+            obj.move_in_sonos_playlist('SQ:0', track=0, new_pos=1)
+
+        Args:
+            object_id (str): The item_id of the Sonos playlist or
+                the Sonos playlist object
+                (:py:class:`~.soco.data_structures.DidlPlaylistContainer`).
+            track (int): **0**-based position of the track to move. The first
+                track is track 0, just like indexing into a Python list.
+            new_pos (int): **0**-based location to move the track.
+            update_id (int): Optional update counter for the object. If left
+                at the default of 0, it will be looked up.
+
+        Returns:
+            tuple: See :py:meth:`reorder_sonos_playlist`
+
+        Raises:
+            SoCoUPnPException: See :py:meth:`reorder_sonos_playlist`
+        """
+        return self.reorder_sonos_playlist(object_id, int(track),
```

Comment:
Yup, it's a helper.  It meant to hand hold.  The definition also specifies an int.